csrc/ir/iostream.cpp

```diff
@@ -80,37 +80,42 @@ void IrPrinter::handle(kir::Kernel& kernel) {
   handle(&kernel);
 }
 
-void IrPrinter::handle(const hir::HostIrContainer* host_fusion) {
-  NVF_CHECK(host_fusion != nullptr);
+void IrPrinter::handle(const hir::HostIrContainer* host_ir_container) {
+  NVF_CHECK(host_ir_container != nullptr);
 
-  // host_fusion declaration
-  os() << "\nHOST FUSION (";
-  for (auto in : host_fusion->inputs()) {
+  // host_ir_container declaration
+  os() << "\nHOST IRS: (";
+  for (auto in : host_ir_container->inputs()) {
     os() << in->toString(indent_size_);
-    if (in != host_fusion->inputs().back()) {
+    if (in != host_ir_container->inputs().back()) {
       os() << ", ";
     }
   }
   os() << ") -> (";
-  for (auto out : host_fusion->outputs()) {
+  for (auto out : host_ir_container->outputs()) {
     os() << out->toString(indent_size_);
-    if (out != host_fusion->outputs().back()) {
+    if (out != host_ir_container->outputs().back()) {
       os() << ", ";
     }
   }
   os() << ") :\n";
 
-  // host_fusion body
+  // host_ir_container body
   indent_size_++;
-  for (auto expr : host_fusion->topLevelExprs()) {
-    os() << expr->toString(indent_size_);
+  for (auto expr : host_ir_container->topLevelExprs()) {
+    os() << expr->toString(indent_size_) << std::endl;
   }
   indent_size_--;
+  for (auto* host_unit : ir_utils::filterByType<hir::HostUnit>(
+           host_ir_container->unordered_exprs())) {
+    os() << std::endl;
+    os() << host_unit->toString(indent_size_);
+  }
   os() << "END.\n\n";
```

Comment:
Nit: the `END.` now looks redundant given the closing curly bracket. 

```
%kernel {
T1_l[ iS3{4}, iS4{8}, iS5{32} ]
   = T0_g[ iS0{4}, iS1{8}, iS2{32} ]
   + T0_g[ iS0{4}, iS1{8}, iS2{32} ];
T2_g[ rS6{4}, iS7{8}, iS8{32} ]
   = reduction( T1_l[ iS3{4}, iS4{8}, iS5{32} ], op = add, initial value = float(0), allreduce = false )
}
END.
```

Reply:
Thanks to @cowanmeg 's comment, I now realize this marks the end of a host IR container instead of a kernel or a host unit. In that case, consider something like

```
%HostIrContainer {
...
}  // %HostIrContainer
```

As Meghan said, it's unclear what `END.` refers to.

Reply:
Right, I agree. I applied your suggestion to `HostIrContainer::print()`, and also for the other print methods, `Fusion::printMath()`, `Fusion::print()`, `Kernel::print()`, to make it uniform.